pennylane/ops/functions/equal.py

```diff
@@ -99,3 +124,30 @@ def equal(
                 return False
 
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
+
+
+def equal_measurements(op1, op2):
+    """Determine whether two MeasurementProcess objects are equal"""
+    return_types_match = op1.return_type == op2.return_type
+    observables_match = op1.obs == op2.obs
```

Comment:
Careful! Using `==` with classes returns `True` only if the class is the same instance. We should do sth like:
```suggestion
    observables_match = qml.equal(op1.obs, op2.obs) if op1.obs is not None else op1.obs == op2.obs
```

Reply:
Thank you for pointing this out, I was not fully aware of it!